### PR TITLE
Poratal2 - get the product price from the price list

### PR DESCRIPTION
**Pull request by arek1-618**
Poratal2 - get the product price from the price list

### CODE REVIEW

File: api/webservice/Portal/Products/RecordsTree.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+			}
+		}
+		$queryGenerator->setCustomColumn('u_#__istorages_products.qtyinstock as storage_qtyinstock');
+		$queryGenerator->addJoin([
+			'LEFT JOIN',
+			'u_#__istorages_products',
+			"u_#__istorages_products.crmid={$this->getUserStorageId()} AND u_#__istorages_products.relcrmid = vtiger_products.productid"]
+		);
+		$queryGenerator->setCustomColumn('a_#__taxes_global.value as tax_value');
+		$queryGenerator->addJoin([
```
Comment (TomaszQr) [inline api/webservice/Portal/Products/RecordsTree.php]: ?

File: api/webservice/Portal/BaseModule/RecordsList.php [hunk 2]
```diff
@@ -92,4 +89,61 @@ public function getQuery()
...
+
+	/**
+	 * Create record from row.
+	 *
+	 * @param array                 $row
+	 * @param \Vtiger_Field_Model[] $fieldsModel
+	 *
+	 * @return array
+	 */
+	protected function createRecordFromRow(array $row, array $fieldsModel): array
```
Comment (rskrzypczak) [inline api/webservice/Portal/BaseModule/RecordsList.php]: ?

File: api/webservice/Portal/BaseModule/RecordsList.php [hunk 3]
```diff
@@ -92,4 +89,61 @@ public function getQuery()
...
+	}
+
+	/**
+	 * Create header.
+	 *
+	 * @param array $fieldsModel
+	 *
+	 * @return array
+	 */
+	protected function createHeader(array $fieldsModel): array
```
Comment (rskrzypczak) [inline api/webservice/Portal/BaseModule/RecordsList.php]: ?

File: api/webservice/Portal/BaseModule/RecordsList.php [hunk 4]
```diff
@@ -92,4 +89,61 @@ public function getQuery()
...
+	}
+
+	/**
+	 * Create raw data from row.
+	 *
+	 * @param array $row
+	 *
+	 * @return array
+	 */
+	protected function createRawDataFromRow(array $row): array
```
Comment (rskrzypczak) [inline api/webservice/Portal/BaseModule/RecordsList.php]: ?

**Pull request merged**